### PR TITLE
Composed Product V2

### PR DESCRIPTION
**Pull request by oxid83**

### CODE REVIEW

File: services/promo.js [hunk 1]
```diff
@@ -172,7 +172,30 @@ const middlewarePromoCatalog = async (req, res) => {
  */
 const checkPromoCatalog = async (products, user = null, lang = null, keepObject = false, populate = [], associatedProducts = false, keepPromos = false) => {
     // TODO : improve speed because it's usefull
-    if (!products || !products.length) return [];
+    if ((!products || !products.length) && (!products || !products.items || !products.items.length)) return [];
```
Comment (haegemonia76) [inline services/promo.js:175]: I think it should be
```js
if ((!products || !products.length) || (!products || !products.items || !products.items.length)) return [];
```
can you explain why you did it that way ?

File: themes/default_theme/pages/cart.js [hunk 2]
```diff
@@ -97,6 +97,7 @@ class PageCart extends NSPageCart {
                                                                         imgAlt     = item.id.images[0].alt || imgAlt;
                                                                     }
                                                                 }
+                                                                console.log(item.selections)
```
Comment (haegemonia76) [inline themes/default_theme/pages/cart.js:100]: remove `console.log`

File: backoffice/assets/translations/fr/bundle.json [hunk 3]
```diff
@@ -32,6 +32,11 @@
         "qt": "Quantité",
         "attrJeux": "Attributs du jeu d'attributs",
         "idFirst": "Vous devez d'abord renseigner l'id du produit avant de pouvoir configurer les attributs !",
-        "catNom": "Nom de la catégorie"
+        "catNom": "Nom de la catégorie",
+        "modifier_price": {
```
Comment (haegemonia76) [inline backoffice/assets/translations/fr/bundle.json:36]: maybe follow camelCase like other properties

File: backoffice/assets/translations/fr/bundle.json [hunk 4]
```diff
@@ -32,6 +32,11 @@
...
+            "ati": "Difference prix TTC ",
+            "et": "Difference prix HT"
+        },
+        "modifier_weight": "Difference de poids (g)"
```
Comment (haegemonia76) [inline backoffice/assets/translations/fr/bundle.json:40]: maybe follow camelCase like other properties

File: backoffice/assets/translations/en/bundle.json [hunk 5]
```diff
@@ -32,6 +32,11 @@
         "qt": "Quantity",
         "attrJeux": "Attributes set attributes",
         "idFirst": "You must type the product's ID before configuring attributes!",
-        "catNom": "Category name"
+        "catNom": "Category name",
+        "modifier_price": {
```
Comment (haegemonia76) [inline backoffice/assets/translations/en/bundle.json:36]: maybe follow camelCase like other properties

File: backoffice/assets/translations/en/bundle.json [hunk 6]
```diff
@@ -32,6 +32,11 @@
...
+            "ati": "ATI price diff",
+            "et": "ET price diff"
+        },
+        "modifier_weight": "Weight diff (gram)"
```
Comment (haegemonia76) [inline backoffice/assets/translations/en/bundle.json:40]: maybe follow camelCase like other properties

**Review by haegemonia76 — CHANGES_REQUESTED** [review]

File: services/promo.js (hunk 1 above)
Comment (oxid83) [reply services/promo.js:183]: The test is correct, we can actually have a products array as 'products' or a cart, so if products.length is undefined, maybe a products.items.legth exists,  so we return [] only if both are not defined or 0

File: orm/schemas/productBundleSchema.js [hunk 7]
```diff
@@ -72,24 +81,48 @@ ProductBundleSchema.methods.addToCart = async function (cart, item) {
...
+                    modifiers.price.et  += itemPrdBundleSectionProduct.modifier_price.et;
+                    modifiers.price.ati += itemPrdBundleSectionProduct.modifier_price.ati;
+                }
+                if (itemPrdBundleSectionProduct && itemPrdBundleSectionProduct.modifier_weight) {
+                    modifiers.weight += itemPrdBundleSectionProduct.modifier_weight;
+                }
+            }
+        }
+    }
+    console.log('Total modifiers: ', modifiers);
```
Comment (haegemonia76) [inline orm/schemas/productBundleSchema.js:125]: remove console.log

**Pull request merged**